.travis.yml

```diff
@@ -5,16 +5,28 @@ go_import_path: github.com/pingcap/tidb
 go:
   - "1.11.3"
 
-env:
-    - TRAVIS_COVERAGE=0
-    - TRAVIS_COVERAGE=1
+cache:
+  directories:
+    - $GOPATH/pkg/mod
+  # TODO: If memory permits, we could cache this dirctory.
+  # - $HOME/.cache/go-build
+
+env: 
```

Comment:
[F.Y.R](https://docs.travis-ci.com/user/speeding-up-the-build#parallelizing-your-builds-across-virtual-machines)
I split the test, and it will speed Travis-CI up.
(Before, we use `make dev upload-coverage`.
And `make upload-coverage` is not used when `TRAVIS_COVERAGE=0`, we can split `make dev` to `make check` and `make test`.
`make test` can also split to `make checkdep`, `make gotest` and `make explaintest`.)

Reply:
If we cache this directory, it will take no effect when we file the PR at first(no speed up).
But when we in the PR push a lot commits, it will cache `gotest` to speed `ravis-CI up.